hyperspy/misc/utils.py

```diff
@@ -1276,51 +1243,15 @@ def guess_output_signal_size(test_signal,
         output_dtype = object
         output_signal_size = ()
     else:
-        output = function(test_signal, **kwargs)
+        # copy.deepcopy is used to avoid the function altering
+        # the original kwargs.
+        output = function(test_signal, **copy.deepcopy(kwargs))
+        output = np.asarray(output)
```

Comment:
Why is `np.asarray` necessary?

Reply:
Since the output from `function` can be basically anything, for example a tuple, the easiest way to get the dtype and shape of the output object is to convert it into a numpy array. Especially since the output from `function` will be put into a numpy array anyway.

```python
import numpy as np
from scipy.ndimage import center_of_mass
output = center_of_mass(np.ones((10, 10))) # (4.5, 4.5)
```
However, there might be other, better, ways of doing this "numpy-conversion".

Reply:
OK, I was questioning this because not all array objects will support `np.asarray`. Maybe something along the line of:
```suggestion
        try:
           output_dtype = output.dtype
        except AttributeError:
           output = np.asarray(output)
           output_dtype = output.dtype
```

Reply:
Good point! Done!